### ISSUE TITLE
Hive: TestHiveIcebergStorageHandler should be slimmed down

### ISSUE THREAD

**Issue by pvary** [apache/iceberg#1826]
On #1607 @rdblue suggested that we should put the Hive specific test to an integration suit like @jackye1995 did with Glue, so if there are changes only in spark or flink code then we do not have to run the whole Hive suite to test the changes.

My understanding of @jackye1995's solution is that we should create a new source folder for the integration tests to separate out the ones we do not want to run every time.
```
  sourceSets {
    integration {
      java.srcDir "$projectDir/src/integration/java"
      resources.srcDir "$projectDir/src/integration/resources"
      compileClasspath += main.output + test.output
      runtimeClasspath += main.output + test.output
    }
  }

  configurations {
    integrationImplementation.extendsFrom testImplementation
    integrationRuntime.extendsFrom testRuntime
  }

  task integrationTest(type: Test) {
    testClassesDirs = sourceSets.integration.output.classesDirs
    classpath = sourceSets.integration.runtimeClasspath
  }
```

What I have not seen or understand is the automation around which would:
1. Decide when we need to run the integration tests
2. Run the actual tests on the CI

Did I miss something? Or this is what should be implemented?
If so, then how should we decide when to run the integration tests? I have found this [thread](https://discuss.gradle.org/t/incremental-testing-support/24785) about it on the gradle forums. And I have seen this kotlin [code](https://medium.com/swlh/incremental-testing-for-gradle-multi-projects-2ad6cacd8e51) as an actual implementation. Also something in [groovy](https://itnext.io/gradle-modules-running-unit-tests-only-in-affected-modules-fff89562339e)
This is what we are looking for? Is Iceberg tests are organized on a way that the subproject compile hierarchy also defines which test should be run? Or we should just stick to the specific spark/flink only change should not start Hive tests?

Thanks,
Peter

**Comment by rdblue** [comment]
> Decide when we need to run the integration tests

I think that we would want to create a new GitHub action that runs Hive integration tests separately and for just one JVM profile. That way, we have an action for JDK8 unit tests, an action for JDK11 unit tests, and a Hive integration test action that runs in parallel with those. That would save overall time by running in parallel, and would save resources by running the expensive tests just once.

I would still want to run the tests on each PR because we don't want changes to core breaking the Hive integration tests. Also, we could possibly run the integration tests with Hive 3 if anything in iceberg-mr or iceberg-hive-metastore changes.

> Run the actual tests on the CI

When we moved to GitHub actions, I updated the python tests to only run when a file in `python/` changes. We could use the same strategy for these tests. But the main thing is moving expensive tests into an `integrationTest` task and calling it from a GitHub action with `./gradlew iceberg-mr:integrationTest -x test`.
